marketplace-and-integrations/packages/authorized-services.md

```diff
@@ -0,0 +1,776 @@
+---
+description: >-
+  Details a package supporting creation of integrations with external services that use an OAuth flow for authentication and authorization.
+---
+
+# Authorized Services
+
+## Introduction
+
+**Umbraco Authorized Services** is an Umbraco package designed to reduce the effort needed to integrate third party solutions into Umbraco solutions.  Many such services require an OAuth flow fir authentication and authorization.  Working with these services requires a fair bit of plumbing code to handle creating an authorized connection.  This is necessary before the developer working with the service can get to using the provided API to implement the business requirements.
+
+There are similarities to the flow that needs to be implemented for different services.  Steps include:
+
+- Redirecting to an authentication endpoint.
+- Handling the response including an authentication code and exchanging it for an access token.
+- Securely storing the token.
+- Including the token in API requests.
+- Serializing requests and deserializing the API responses.
+- Handling cases where the token has expired and obtaining a new one via a refresh token.
+
+There are though also differences, across request and response structures and variations in the details of the flow itself.
+
+The package tries to implement a single, best practice implementation of working with OAuth. For particular providers the specific flow required can be customized via configuration or code.
+
+## Features
+
+For the solution developer, the Umbraco Authorized Services offers two primary features.
+
+Firstly there's a tree available in the _Settings_ section of the backoffice, called _Authorized Services_. The tree shows the list of services based on the details provided in configuration.
```

Comment:
Suggestion to add sub-headings in this section - it will highlight the 2 features a little better.

```suggestion
### Authorized Services in the backoffice

Firstly there's a tree available in the _Settings_ section of the backoffice, called _Authorized Services_. The tree shows the list of services based on the details provided in configuration.
```